server/core_purchase.go

```diff
@@ -410,11 +406,15 @@ returning transaction_id, create_time, update_time"
 	if err != nil {
 		return nil, err
 	}
+	defer rows.Close()
```

Comment:
There's already a `rows.Close()` after the loop so I would just move this into the loop ahead of the error return, rather than as a `defer` here.